alternator/executor.cc

```diff
@@ -2191,16 +2191,28 @@ future<executor::request_return_type> executor::batch_write_item(client_state& c
                 auto mut_key = std::make_pair(mutation_builders.back().second.pk(),
                         mutation_builders.back().second.ck());
                 if (used_keys.contains(mut_key)) {
-                    return make_ready_future<request_return_type>(api_error::validation("Provided list of item keys contains duplicates"));
+                    co_return api_error::validation("Provided list of item keys contains duplicates");
                 }
                 used_keys.insert(std::move(mut_key));
             } else {
-                return make_ready_future<request_return_type>(api_error::validation(format("Unknown BatchWriteItem request type: {}", r_name)));
+                co_return api_error::validation(format("Unknown BatchWriteItem request type: {}", r_name));
             }
         }
     }
 
-    return do_batch_write(_proxy, _ssg, std::move(mutation_builders), client_state, trace_state, std::move(permit), _stats).then([] () {
+    for (const auto& b : mutation_builders) {
+        // TODO: if check_has_permission is slower than a simple hash-table
+        // lookup, it makes sense to build a set of different schemas and
+        // only run check_has_permission on each schema once.
```

Comment:
It should be fast enough to avoid the extra work to deduplicate the checks.

Reply:
Ok, I'll remove the TODO then.